### PR TITLE
[dist] fix: keep FSDP2 buffers on compute device with CPU offload

### PR DESCRIPTION
**Pull request by YLlllllllllll**
## Summary

Keep registered model buffers on the rank-local compute device when FSDP2 parameter CPU offload is enabled.

FSDP2 CPU offload still owns the parameter, reduced-gradient, and optimizer-state residency contract. This change avoids moving compute-time buffers such as rotary state to CPU, where they can cause device-mismatch failures during forward.

## What changed

- Move registered persistent and non-persistent buffers to the current rank's compute device after FSDP2 materialization and weight loading, without moving parameters.
- Preserve buffer alias groups when moving them.
- Preserve buffer aliases across the skip-HF DCP materialization path:
  - groups containing only non-persistent buffers retain their config-derived values;
  - groups with any persistent registration are populated by the subsequent DCP restore.
- Log the number and bytes of buffers moved.
- Add coverage for plain tensors, DTensor aliases, gradients, lazy AdamW moments, HF loading, DCP save/load, and fresh cold-resume construction.

## Relationship to other work

#776 addresses a Qwen3.5 model-specific instance of GPU buffer placement. This PR implements the generic FSDP2 distributed-loading fix and does not modify model-specific files.

Activation offload remains orthogonal: synchronous and asynchronous activation modes are covered by #964 and #965 respectively.

## Verification

- focused CPU alias/DCP tests: passed
- focused offload test set: CPU cases passed; the accelerator residency case skipped correctly without two visible devices
- `python -m py_compile` on changed modules/tests: passed
- `make quality`: passed
- `git diff --check`: passed
- independent VeOmni review: `safe` after three review rounds

The added hardware gate requires at least two visible accelerator devices and exercises real FSDP2 forward/backward, CPU gradients, AdamW moment residency, DCP save, and fresh cold resume.

## Hardware status

The two-device CUDA/NPU gate was not executed on the review host. Ascend 910B4-1 validation is intentionally tracked as the next integration step; CPU/gloo results are not claimed as accelerator device-split evidence.

This PR contains no private Mojo backend, Byted dataloader, HDFS path, or experiment-only keepalive code.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved CPU-offloaded training and checkpoint resume behavior.
  - Preserved model buffers, shared buffer values, and derived runtime state across checkpoint restoration.
  - Ensured restored models produce consistent outputs without unnecessary reinitialization.
  - Improved device selection and buffer placement for accelerator-based training.

- **Tests**
  - Added regression coverage for GPU, NPU, and CPU-offload workflows, including distributed training and checkpoint restoration.

### CODE REVIEW

**Comment by YLlllllllllll** [comment]
@Luosuu All CI checks passed on `ff45cc7` (3 green) — ready to merge when you have a moment.

**Comment by YLlllllllllll** [comment]
@cursor[bot] The requested review fixes are now pushed to this PR's updated head. Please continue the review against the new head, with particular attention to the previously reported issues.

**Comment by YLlllllllllll** [comment]
@cursor[bot] Merge conflicts with upstream main are resolved, and the final reviewed fixes are now on the new head. Please re-review this updated head, especially the previously reported issues.

**Comment by YLlllllllllll** [comment]
@cursor Supplemental status against the current head `5fc05b9`:

`tests/checkpoints/test_dcp_checkpointer.py` is now explicitly run in both the GPU and NPU unit-test workflows, and the CPU regression covers the buffer-alias/resume path without requiring two accelerators. The DTensor byte accounting now uses the local shard.

The remaining review item is explanatory/coverage-related: two pre-existing tests are intentionally skipped on CPU because they exercise accelerator-only behavior, and the multi-device path should be confirmed from the CI report rather than inferred from a green job. We will add the rationale/coverage evidence; this is not currently a correctness change. Please review the current head rather than the pre-sync baseline.

**Comment by YLlllllllllll** [comment]
Merged current `main` at `8eff52ed5` and resolved `veomni/distributed/torch_parallelize.py`.

Conflict resolution:
- keep this PR's buffer alias-group rebinding (`to_empty()` otherwise splits shared registrations)
- absorb main's meta-buffer warning and `copy_` into the materialized tensor
- snapshot only all-nonpersistent groups; mixed persistent/nonpersistent groups stay checkpoint-authoritative

CPU skip reasons now point at `tests/checkpoints/test_dcp_checkpointer.py` (already in both GPU and NPU unit workflows). The two-device residency test stays in `tests/utils/test_rank0_load_and_broadcast_weights.py` and self-skips unless two accelerators are present.

@cursor please re-review this head.

File: .github/workflows/gpu_unit_tests.yml [hunk 1]
```diff
@@ -216,3 +216,6 @@ jobs:
           uv run --frozen pytest -s -x tests/lora/test_moe_lora_trainer.py
           uv run --frozen pytest -s -x tests/lora/test_moe_lora_ep2.py
           uv run --frozen pytest -s -x tests/lora/test_moe_lora_ep_sharded_stream_load.py
+      - name: Run DCP checkpointer CPU-offload regression tests
+        run: |
+          uv run --frozen pytest -s -x tests/checkpoints/test_dcp_checkpointer.py
```
[the remark this answers is not in the document]
Comment (YLlllllllllll) [reply .github/workflows/gpu_unit_tests.yml]: Fixed on `601ec9940`. Removed the duplicate GPU/NPU workflow steps. `tests/checkpoints/test_dcp_checkpointer.py` already runs in the earlier checkpoint-callback step, including `test_cpu_offload_dcp_resume_restores_persistent_and_nonpersistent_buffers`.

**Comment by YLlllllllllll** [comment]
Addressed the CodeRabbit finding on `601ec9940`: the extra DCP checkpointer workflow steps are gone so that file runs once per GPU/NPU job. Please re-review this head.
